pom.xml

```diff
@@ -122,6 +122,43 @@
                 </plugins>
             </build>
         </profile>
+
+        <!-- Use -X option to see classpath -->
+        <profile>
+            <id>jnr-exclude</id>
+            <!-- Compile main and test code with jnr-unixsocket in classpath, but drop it when
+                 running tests. Make sure UDP and Windows support works even if jnr-unixsocket
+                 dependency is not available.
+            -->
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <version>2.19</version>
+                        <configuration>
+                            <classpathDependencyExcludes>
+                                <classpathDependencyExclude>com.github.jnr:jnr-unixsocket</classpathDependencyExclude>
+                            </classpathDependencyExcludes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>jnr-latest</id>
+            <!-- This profile is for testing for compatibility with the latest version of
+                 jnr-unixsocket. Maven does not rebuild main or test code when profile changes,
+                 so we can test how the code compiled with version X runs with version Y.
+            -->
+            <dependencies>
+                 <dependency>
+                    <groupId>com.github.jnr</groupId>
+                    <artifactId>jnr-unixsocket</artifactId>
+                    <version>0.38.15</version>
```

Comment:
I believe this version is not java7 compatible and per: https://github.com/DataDog/java-dogstatsd-client/commit/eed7a1cdb4f3ee17ccffc2409f4000475cc825b2, it appears to be the case. Let's check.

Reply:
It isn't, and we still depend on 0.36. This bit is in a profile that is only activated for testing. The idea is to validate the library using both 0.36 and 0.38.15, so we can provide a bit of assurance for users that don't need Java 7 compatibility and would like a later version.

Reply:
Gotcha!